### PR TITLE
[SYCL] On windows with ms compatibility version, don't add GNU macros

### PR DESCRIPTION
**Pull request by mibintc**
Signed-off-by: Blower <melanie.blower@intel.com>

### CODE REVIEW

**Comment by mibintc** [comment]
Yes I think that’s right.  Shall I poke it in now? MikeT what do you think?

**Comment by mdtoguchi** [comment]
I don't have objections to adding it, especially given what the documentation states.

**Comment by v-klochkov** [comment]
If you add -fdelayed-template-parsing for SYCL compilation, then it would be useful to also add and a simple test case too.
The test should fail without before the patch (i.e. without adding -fdelayed-template-parsing) and pass with it, or/and a check that driver includes that switch when compiles with -fsycl.

I suppose there should be some available tests already to give inspiration for a new test, grep for 'delayed-template-parsing'.

**Comment by mibintc** [comment]
I don’t have time to add a test right this minute, can I put it on my list of things to do, or shall we pull it out.

**Comment by v-klochkov** [comment]
BTW, excuse me for editing 2 of your comments above. I did that to avoid having huge reply messages including aux info about e-mails, dates, about notifications why that e-mail was sent to you, etc.

GitHub has a pretty convenient tool to quote someone's message. To do that you can click to '...' symbol at the right/top corner of someone's message and click 'Quote reply'.

File: clang/test/SemaSYCL/msvc-macro.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// RUN: %clang -fsycl -c %s
```
Comment (bader) [inline clang/test/SemaSYCL/msvc-macro.cpp]: I think the check should be done differently.
Ideally, we should do cross-compilation for Windows, currently test detects the native OS, so people running this test on Linux won't be able to detect regressions.

Probably it's better to move pre-processor tests. E.g. improve this test: https://github.com/intel/llvm/blob/sycl/clang/test/Preprocessor/sycl-macro.cpp

```
RUN: %clang_cc1 %s -fsycl-is-device -E -dM -aux-triple=...-windows-msvc | FileCheck --check-prefix=CHECK-MSVC %s

CHECK-MSVC-NOT: __GNUC__
CHECK-MSVC-NOT: __STDC__
```

**Comment by mibintc** [comment]
@premanandrao the fix to the program is OK but test cases need improvement

**Comment by bader** [comment]
> @premanandrao the fix to the program is OK but test cases need improvement

@mibintc, @premanandrao will you update the test?

**Comment by premanandrao** [comment]
> @mibintc, @premanandrao will you update the test?

I was on vacation.   Will take care of this soon.

**Pull request closed without merging**